src/version.h

```diff
@@ -1 +1,13 @@
-#define REDIS_VERSION "999.999.999"
+#define xstr(s) str(s)
+#define str(s) #s
+
+/*
+ * Version are at the form of 0x00MMmmpp
+ * So each number is at most 1 byte, this means that the max
+ * value for each number is 255.
+ */
+#define REDIS_MAJOR_VERSION 255
+#define REDIS_MINOR_VERSION 255
+#define REDIS_PATCH_VERSION 255
+
+#define REDIS_VERSION xstr(REDIS_MAJOR_VERSION) "."  xstr(REDIS_MINOR_VERSION) "." xstr(REDIS_PATCH_VERSION)
```

Comment:
i had in mind something a lot simpler:
```
#define REDIS_VERSION "255.255.255"
#define REDIS_VERSION_NUM 0x00ffffff
```

it does mean i'll have to update two lines rather than one when releasing a new version, but i'm not sure there's any use for that much detail (like specific defines for each part). no one's gonna use it (for now)
what do you think?

Reply:
@yossigo @itamarhaber it turns out some modules use the following formula for their version numbers:
```
int version = major*10000 + minor*100 + patch;
```
using `%` to extract the individual gives me nausea, but maybe it's just me.
also, this isn't documented anywhere and doesn't bind redis.
what do you think?

Reply:
> i had in mind something a lot simpler:
> 
> ```
> #define REDIS_VERSION "255.255.255"
> #define REDIS_VERSION_NUM 0x00ffffff
> ```
> 
> it does mean i'll have to update two lines rather than one when releasing a new version, but i'm not sure there's any use for that much detail (like specific defines for each part). no one's gonna use it (for now)
> what do you think?

I prefer the update once approach but I can change it if you like.